packages/juce-blueprint/src/components/Canvas.ts

```diff
@@ -0,0 +1,116 @@
+import React, { Component } from 'react';
+
+export function bindCanvasContextProperties(ctx: any) {
+  Object.defineProperty(ctx, 'fillStyle', {
+    enumerable: false,
+    configurable: false,
+    get: function() {
+      return 'Not Supported';
+    },
+    set: function(value) {
+      this.__setFillStyle(value);
+    }
+  });
+
+  Object.defineProperty(ctx, 'strokeStyle', {
+    enumerable: false,
+    configurable: false,
+    get: function() {
+      return 'Not Supported';
+    },
+    set: function(value) {
+      this.__setStrokeStyle(value);
+    }
+  });
+
+  Object.defineProperty(ctx, 'lineWidth', {
+    enumerable: false,
+    configurable: false,
+    get: function() {
+      return 'Not Supported';
+    },
+    set: function(value) {
+      this.__setLineWidth(value);
+    }
+  });
+
+  Object.defineProperty(ctx, 'font', {
+    enumerable: false,
+    configurable: false,
+    get: function() {
+      return 'Not Supported';
+    },
+    set: function(value) {
+      this.__setFont(value);
+    }
+  });
+
+  Object.defineProperty(ctx, 'textAlign', {
+    enumerable: false,
+    configurable: false,
+    get: function() {
+      return 'Not Supported';
+    },
+    set: function(value) {
+      this.__setTextAlign(value);
+    }
+  });
+}
+
+export interface CanvasProps {
+  onMeasure?: (width: number, height: number) => void;
+  onDraw: (ctx: CanvasRenderingContext2D) => void;
+  autoclear?: boolean;
+}
+
+interface CanvasState {
+  width: number;
+  height: number;
+}
+
+export class Canvas extends Component<CanvasProps, CanvasState> {
+  constructor(props: CanvasProps) {
+    super(props);
+
+    this._onMeasure = this._onMeasure.bind(this);
+    this._onDraw = this._onDraw.bind(this);
+
+    this.state = {
+      width: 0,
+      height: 0
+    };
+  }
+
+  _onMeasure(e: any) {
```

Comment:
This _onMeasure callback  was still `_onMeasure(width, height)` which caused some issues in my app after the SyntheticEvents change so I've just fixed as part of this PR.

Reply:
Added in https://github.com/nick-thompson/blueprint/pull/99/commits/3d2fe005116dfdb14b58f2a8afe2d704f0faeb3d. Let me know what you think @nick-thompson

Reply:
It would be great to type `View` props and avoid having to replicate `onMeasure` everywhere so that this sort of thing can be trapped a bit more easily.